### PR TITLE
networking: rename shadow variables

### PR DESCRIPTION
**Pull request by dcpleung**
This is an attempt to rename shadow variables found by enabling -Wshadow manually via twister. Note that this does not enable -Wshadow by default, and that requires more work to be done not only in tree but also in various modules.

This covers mainly networking related code.

Relates to #5450

### CODE REVIEW

File: subsys/net/ip/net_shell.c [hunk 1]
```diff
@@ -2974,14 +2974,14 @@ static void gptp_print_port_info(const struct shell *sh, int port)
 	struct gptp_port_bmca_data *port_bmca_data;
 	struct gptp_port_param_ds *port_param_ds;
 	struct gptp_port_states *port_state;
-	struct gptp_domain *gptp_domain;
+	struct gptp_domain *one_gptp_domain;
```
[the remark this answers is not in the document]
Comment (dcpleung) [reply subsys/net/ip/net_shell.c]: Rename to `domain`.

File: tests/net/ieee802154/l2/src/ieee802154_shell_test.c [hunk 2]
```diff
@@ -25,7 +25,7 @@ extern struct net_pkt *current_pkt;
 extern struct k_sem driver_lock;
 extern uint8_t mock_ext_addr_be[8];
 
-static struct net_if *iface;
+static struct net_if *net_iface;
```
Comment (fgrandel) [inline tests/net/ieee802154/l2/src/ieee802154_shell_test.c:28]: Wow, this was a lot of work. Many thanks! The comment from Andy Ross (optionally) applies here as well: https://github.com/zephyrproject-rtos/zephyr/pull/61234#pullrequestreview-1566078112

**Review by ghost — DISMISSED** [review]

**Review by jukkar — APPROVED** [review]
LGTM

**Review by nandojve — APPROVED** [review]

**Review by rlubos — APPROVED** [review]
LGTM

**Review by SeppoTakalo — APPROVED** [review]

**Pull request merged**